### PR TITLE
feat: add new pull request template

### PR DESCRIPTION
**Pull request by leseb**
This commit introduces a new template for pull requests. The template includes sections for a description of the changes, the issue number being resolved, and a checklist for commit message formatting, updating release notes, and adding necessary documentation and tests.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
DEPENDS ON https://github.com/instructlab/instructlab/pull/1231 for the `PendingReleaseNotes.md` part.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+**Issue resolved by this Pull Request:**
 Resolves #
+--->
+
+**Checklist:**
 
-**Description of your changes:**
+- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
+  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
+- [ ] [Pending release notes](https://github.com/instructlab/instructlab/blob/main/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
```
Comment (russellb) [inline .github/PULL_REQUEST_TEMPLATE.md:18]:
```suggestion
- [ ] [Release Notes](https://github.com/instructlab/instructlab/blob/main/RELEASE_NOTES.md) updated with breaking and/or notable changes for the next minor release.
```

**Review by russellb — CHANGES_REQUESTED** [review]
One minor change needed based on how the release notes PR evolved

As a side note, one thing I tried in another project to help enforce these checklist items was use labels. It's pretty noisy on label usage, but it's a way to prevent merging until the checklist is evaluated.

Merge criteria includes:

```yaml
      - or:
        - label=has-design
        - label=no-design-required
      - or:
        - label=has-docs
        - label=no-docs-required
      - or:
        - label=has-tests
        - label=no-tests-required
```

and then some extra label automation:

```yaml
  # Give a hint via label when no docs label has been applied
  - name: Label when docs label is missing
    conditions:
      - -label=has-docs
      - -label=no-docs-required
      - -author=dependabot[bot]
      - -draft
    actions:
      label:
        add:
          - missing-docs-label
  - name: Remove label when docs label is present
    conditions:
      - or:
        - label=has-docs
        - label=no-docs-required
        - author=dependabot[bot]
        - draft
    actions:
      label:
        remove:
          - missing-docs-label

  # Give a hint via label when no design doc label has been applied
  - name: Label when design doc label is missing
    conditions:
      - -label=has-design
      - -label=no-design-required
      - -author=dependabot[bot]
      - -draft
    actions:
      label:
        add:
          - missing-design-label
  - name: Remove label when design doc label is present
    conditions:
      - or:
        - label=has-design
        - label=no-design-required
        - author=dependabot[bot]
        - draft
    actions:
      label:
        remove:
          - missing-design-label

  # Give a hint via label when no tests label has been applied
  - name: Label when tests label is missing
    conditions:
      - -label=has-tests
      - -label=no-tests-required
      - -author=dependabot[bot]
      - -draft
    actions:
      label:
        add:
          - missing-tests-label
  - name: Remove label when tests label is present
    conditions:
      - or:
        - label=has-tests
        - label=no-tests-required
        - author=dependabot[bot]
        - draft
    actions:
      label:
        remove:
          - missing-tests-label
```

**Comment by leseb** [comment]
> One minor change needed based on how the release notes PR evolved
> 
> As a side note, one thing I tried in another project to help enforce these checklist items was use labels. It's pretty noisy on label usage, but it's a way to prevent merging until the checklist is evaluated.
> 
> Merge criteria includes:
> 
> ```yaml
>       - or:
>         - label=has-design
>         - label=no-design-required
>       - or:
>         - label=has-docs
>         - label=no-docs-required
>       - or:
>         - label=has-tests
>         - label=no-tests-required
> ```
> 
> and then some extra label automation:
> 
> ```yaml
>   # Give a hint via label when no docs label has been applied
>   - name: Label when docs label is missing
>     conditions:
>       - -label=has-docs
>       - -label=no-docs-required
>       - -author=dependabot[bot]
>       - -draft
>     actions:
>       label:
>         add:
>           - missing-docs-label
>   - name: Remove label when docs label is present
>     conditions:
>       - or:
>         - label=has-docs
>         - label=no-docs-required
>         - author=dependabot[bot]
>         - draft
>     actions:
>       label:
>         remove:
>           - missing-docs-label
> 
>   # Give a hint via label when no design doc label has been applied
>   - name: Label when design doc label is missing
>     conditions:
>       - -label=has-design
>       - -label=no-design-required
>       - -author=dependabot[bot]
>       - -draft
>     actions:
>       label:
>         add:
>           - missing-design-label
>   - name: Remove label when design doc label is present
>     conditions:
>       - or:
>         - label=has-design
>         - label=no-design-required
>         - author=dependabot[bot]
>         - draft
>     actions:
>       label:
>         remove:
>           - missing-design-label
> 
>   # Give a hint via label when no tests label has been applied
>   - name: Label when tests label is missing
>     conditions:
>       - -label=has-tests
>       - -label=no-tests-required
>       - -author=dependabot[bot]
>       - -draft
>     actions:
>       label:
>         add:
>           - missing-tests-label
>   - name: Remove label when tests label is present
>     conditions:
>       - or:
>         - label=has-tests
>         - label=no-tests-required
>         - author=dependabot[bot]
>         - draft
>     actions:
>       label:
>         remove:
>           - missing-tests-label
> ```

Can we revisit the labels in a followup discussion? IMO this is a good enough as a starter and better than what we have right now. @russellb thoughts?

**Comment by russellb** [comment]
> Can we revisit the labels in a follow-up discussion? IMO this is good enough as a starter and better than what we have right now. @russellb thoughts?

Absolutely. It was more just an "FYI" kind of thing of a related mergify config I had sitting around. I don't care to add it right away or anything.

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**